### PR TITLE
Cherry pick git no confict resolution

### PR DESCRIPTION
**Pull request by tidy-dev**
Part of work on #1685
## Description

Adds basic cherry pick git logic without conflict resolution

Notes: no-notes

### CODE REVIEW

File: app/src/lib/git/cherry-pick.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const result = await git(
+    ['cherry-pick', revisionRange],
+    repository.path,
+    'cherry pick'
+  )
+
+  return parseCherryPickResult(result)
+}
+
+function parseCherryPickResult(result: IGitResult): CherryPickResult {
```
Comment (sergiou87) [inline app/src/lib/git/cherry-pick.ts:39]: I'm sure you already know, but just in case: you can probably use one of the new parsers introduced in #11486

File: app/test/helpers/repository-builder-cherry-pick-test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { Repository } from '../../src/models/repository'
+import { setupEmptyRepositoryDefaultMain } from './repositories'
+import { makeCommit, switchTo, createBranch } from './repository-scaffolding'
+
+/**
+ * Creates a test repository to be used for testing cherry pick behaviour with:
+ *  - one commits on default branch,
```
Comment (sergiou87) [inline app/test/helpers/repository-builder-cherry-pick-test.ts]: Nit: typo
```suggestion
 *  - one commit on default branch,
```

File: app/test/unit/git/cherry-pick-test.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+      featureBranch.tip.sha
+    )
+
+    const commits = await getCommits(repository, targetBranch.ref, 3)
+    expect(commits.length).toBe(2)
+    expect(commits[0].summary).toBe(cherryPickedCommit!.summary)
+    expect(result).toBe(CherryPickResult.CompletedWithoutError)
+  })
+
+  it('successfully cherry pick a commit with empty message', async () => {
```
Comment (sergiou87) [inline app/test/unit/git/cherry-pick-test.ts:59]: Is this correct? I just learned cherry-picking commits with empty messages was a special scenario, so I looked into the documentation and seems like we need a `--allow-empty-message` parameter when calling `git cherry-pick`: https://git-scm.com/docs/git-cherry-pick#Documentation/git-cherry-pick.txt---allow-empty-message

However, that parameter is not passed to `git cherry-pick` in this PR. Am I missing anything? 😳

File: app/test/unit/git/cherry-pick-test.ts [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+          contents: '# HELLO WORLD! \nTHINGS GO HERE\n',
+        },
+      ],
+    }
+    await makeCommit(repository, featureBranchCommitThree)
+
+    featureBranch = await getBranchOrError(repository, featureBranchName)
+    await switchTo(repository, targetBranchName)
+
+    const commitRange = `${firstCommitSha}^..${featureBranch.tip.sha}`
```
Comment (sergiou87) [inline app/test/unit/git/cherry-pick-test.ts]: Not asking you to do anything here, but I'm just wondering if we should add a function for this like `revRange` and `revSymmetricDifference` that we have in `app/src/lib/git/rev-list.ts`.

That specific notation is only used in this file so 🤷‍♂️ I'll leave it up to you 😄

**Review by sergiou87 — COMMENTED** [review]
Great work!! I love those tests 😍 

I only have a nit that you can ignore, a suggestion that you can ignore (😂 ) and a question about one of your tests.

File: app/src/lib/git/cherry-pick.ts (hunk 1 above)
Comment (tidy-dev) [reply app/src/lib/git/cherry-pick.ts:39]: Thanks for the reminder! I  believe that may be more prudent when I add the progress callback in - there is a bit more parsing on that - it pulls a few more attributes out to monitor progress. All this method is responsible for is mapping the dugite/git exit/error code to a cherry pick result (following the pattern established in rebase) to allow for the app store to translate that into a user friendly response. But, I will keep it in mind in case there is better way to do both now :)

File: app/test/unit/git/cherry-pick-test.ts (hunk 3 above)
Comment (tidy-dev) [reply app/test/unit/git/cherry-pick-test.ts:59]: Nope. I was also a little perplexed by that. (I actually still have it in my write up to the contrary) I originally had it grouped with the failures. But, I confirmed that this is adding a commit with an empty message and it does in fact cherry pick it. Thus, I am not sure why the documentation indicates that it would fail.

File: app/test/unit/git/cherry-pick-test.ts (hunk 4 above)
Comment (tidy-dev) [reply app/test/unit/git/cherry-pick-test.ts]: :D - I was going to look to see if we had something like this when I went to implement range cherry picking ui.  I will go ahead and add it since I will need for that later PR anyways.

File: app/test/unit/git/cherry-pick-test.ts (hunk 3 above)
Comment (tidy-dev) [reply app/test/unit/git/cherry-pick-test.ts:59]: I just added a couple more lines to the test to verify before cherry picking that the commit I am cherry picking does have an empty message. 🤷‍♀️

Comment (sergiou87) [reply app/test/unit/git/cherry-pick-test.ts:59]: Hmmm maybe the behavior changed in a more recent version of git 🤔
Anyway, that additional check you just added is 💯 , thank you!

**Review by sergiou87 — APPROVED** [review]
LGTM, thanks for those changes! 😍 :shipit:
